crucible-mir/src/Mir/Trans.hs

```diff
@@ -149,33 +149,64 @@ transConstVal _ty (Some (UsizeRepr)) (M.ConstInt i) =
        return $ MirExp UsizeRepr (S.app $ usizeLit n)
 transConstVal _ty (Some (IsizeRepr)) (ConstInt i) =
       return $ MirExp IsizeRepr (S.app $ isizeLit (fromIntegerLit i))
-transConstVal (M.TyRef (M.TySlice ty) _) (Some (MirSliceRepr tpr)) (M.ConstSlice cs) = do
-    cs' <- Trav.for cs $ \c -> do
-        MirExp tpr' c' <- transConstVal ty (Some tpr) c
-        Refl <- testEqualityOrFail tpr tpr' $
-            "transConstVal (ConstSlice): returned wrong type: expected " ++
-            show tpr ++ ", got " ++ show tpr'
-        pure c'
-    vec <- mirVector_fromVector tpr $ R.App $ E.VectorLit tpr $ V.fromList cs'
-    vecRef <- constMirRef (MirVectorRepr tpr) vec
-    ref <- subindexRef tpr vecRef (R.App $ usizeLit 0)
-    let len = R.App $ usizeLit $ fromIntegral $ length cs
-    let struct = S.mkStruct
-            (mirSliceCtxRepr tpr)
-            (Ctx.Empty Ctx.:> ref Ctx.:> len)
-    return $ MirExp (MirSliceRepr tpr) struct
-transConstVal _ty (Some (MirSliceRepr u8Repr@(C.BVRepr w))) (M.ConstStr bs)
+
+--
+-- Before the slice code got split into separate body and reference steps,
+-- it did the following after the code that's equivalent to the ConstArray
+-- case (which is where constant slice bodies get handled now):
+--
+--vecRef <- constMirRef vec_tpr vec
+--ref <- subindexRef tpr vecRef (R.App $ usizeLit 0)
+--let len = R.App $ usizeLit $ fromIntegral $ length cs
+--let struct = S.mkStruct (mirSliceCtxRepr tpr) (Ctx.Empty Ctx.:> ref Ctx.:> len)
+--return $ MirExp (MirSliceRepr tpr) struct
```

Comment:
I think it's fine to include a historical note about the previous approach used to translate constant slices, but I worry that including the full code (verbatim) will be a bit much if you don't have the full context for what all of these variables are. (For instance, `vec` no longer appears in the new version of the code—I think you'd be better off going through the `git` history if you want the previous implementation.)

My personal preference would be to:

* Describe the steps needed to turn the looked-up `DefId` into a `MirExp` (as you currently do in the bulleted list below).
* Note that MIR slice references used to be translated in a way such that they would be turned into references to constant vectors, but this prevented them from being eligible for equality or overlap checks (citing https://github.com/GaloisInc/saw-script/issues/2064). The current approach instead translates slice references into references to separate global allocations (citing the `ConstStrBody` body case below).

Reply:
That comment's there as a semi-proof to myself that I haven't accidentally changed the behavior.

There's a lot of things here where you can write the same action as either ```a``` or ```A``` (so to speak) and then assorted combinations of ```Ab```, ```Bc```, ```bC```, etc. such that if you started with ```((Ab,cD),ef)``` and want to change it to ```(a, (bCd), EF)``` or similar you want to be pretty careful that what you've changed it to is actually equivalent, and neither missing a step nor replacing ```c``` with ```C'``` instead of ```C``` when you didn't mean to. The first couple times I got it wrong (even laying it out explicitly), hence the high level of paranoia.

Now that it's running maybe it should just be flushed, though.

Let me chew on it over the weekend.

Reply:
Conclusion: it should just be flushed. It is only of interest if the new code turns out not to work in some corner case, and in that case fetching things out of history is fine.